crates/syntax/src/ast/make.rs

```diff
@@ -158,6 +160,52 @@ fn ty_from_text(text: &str) -> ast::Type {
     ast_from_text(&format!("type _T = {text};"))
 }
 
+/** Related goto [link](https://doc.rust-lang.org/reference/items/type-aliases.html)
+    Type Alias syntax is
+
+    ```
+    TypeAlias :
+       type IDENTIFIER GenericParams? ( : TypeParamBounds )? WhereClause? ( = Type WhereClause?)? ;
+    ```
+
+    FIXME : ident should be of type ast::Ident
+*/
+pub fn ty_alias(
+    ident: String,
+    generic_param_list: Option<ast::GenericParamList>,
+    type_param_bounds: Option<ast::TypeParam>,
+    where_clause: Option<WhereClause>,
+    assignment: Option<(ast::Type, Option<ast::WhereClause>)>,
+) -> ast::TypeAlias {
+    let mut s = String::new();
+    s.push_str(format!("type {}", ident.as_str()).as_str());
+
+    if let Some(list) = generic_param_list {
+        s.push_str(list.to_string().as_str());
```

Comment:
```suggestion
        s.push_str(list.to_string());
```
These `as_str` calls shouldn't be needed, as `String` itself implements `Display` already